### PR TITLE
Replace developerError with Check in polygonGeometry

### PR DESCRIPTION
**Pull request by rahwang**
Addressing #4794

### CODE REVIEW

File: Source/Core/PolygonGeometry.js [hunk 1]
```diff
@@ -651,9 +652,7 @@ define([
         options = defaultValue(options, defaultValue.EMPTY_OBJECT);
 
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(options.positions)) {
-            throw new DeveloperError('options.positions is required.');
-        }
+        Check.typeOf.object('options.positions', options.positions);
```
Comment (lilleyse) [inline Source/Core/PolygonGeometry.js]: Use `Check.defined`.

File: Source/Core/PolygonOutlineGeometry.js [hunk 2]
```diff
@@ -444,9 +440,7 @@ define([
         options = defaultValue(options, defaultValue.EMPTY_OBJECT);
 
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(options.positions)) {
-            throw new DeveloperError('options.positions is required.');
-        }
+        Check.typeOf.object('options.positions', options.positions);
```
Comment (lilleyse) [inline Source/Core/PolygonOutlineGeometry.js]: Same here.

Comment (rahwang) [inline Source/Core/PolygonOutlineGeometry.js]: Done!

**Comment by lilleyse** [comment]
Thanks! The build failure seems to be unrelated to this PR.

**Pull request merged**